### PR TITLE
Display Profile Picture + Move Microsoft Graph into Redux

### PR DESCRIPTION
**Pull request by kodyang**
## Description
This pr adds profile pictures associated with microsoft account. It also moves the other api calls to microsoft graph into redux, and saves the access token into the store.
[AB#33705](https://microsoftgarage.visualstudio.com/002806e2-ebaa-4672-9d2e-5fe5d29154ef/_workitems/edit/33705)

## Screenshots
No profile picture:
image

Profile picture:
image

### CODE REVIEW

File: private-templates-service/client/src/store/auth/actions.ts [hunk 1]
```diff
@@ -1,16 +1,156 @@
...
+function requestProfilePictureFailure(): GetProfilePictureAction {
+  return {
+    type: GET_PROFILE_PICTURE_FAILURE
+  }
+}
+
+export function getUserDetails() {
+  return function (dispatch: any, getState: () => RootState) {
+    dispatch(requestUserDetails());
+    const state = getState();
```
Comment (guptaharsh25) [inline private-templates-service/client/src/store/auth/actions.ts]: nice, didn't know this was possible

Comment (kodyang) [reply private-templates-service/client/src/store/auth/actions.ts]: took me a while to figure it out haha

File: private-templates-service/client/src/components/SideBar/UserAvatar.tsx [hunk 2]
```diff
@@ -15,21 +15,23 @@ const mapStateToProps = (state: RootState) => {
...
-export function UserAvatar(props: UserAvatarProps): ReactElement {
+class UserAvatar extends React.Component<Props> {
   // If a user avatar is available, return an img tag with the pic
-  if (props.user && props.user.avatar) {
+  render() {
+    if (this.props.user && this.props.user.imageURL) {
+      return (
+        <AvatarIcon
+          src={this.props.user.imageURL}
+          alt="user"
```
Comment (nnmkhang) [inline private-templates-service/client/src/components/SideBar/UserAvatar.tsx]: Should we have a more descriptive alt text for accessibility?

**Comment by kodyang** [comment]
didn't mean to re-request approval, sorry Rachel!

**Pull request merged**